tests/compute/test_sampler.py

```diff
@@ -283,8 +287,13 @@ def check_negative_sampler(mode, exclude_positive, neg_size):
         else:
             assert F.array_equal(g.has_edges_between(neg_src, neg_dst), exist)
 
+        total_samples += batch_size
+        if (total_samples >= max_samples):
```

Comment:
Why do we need this check?

Reply:
The weighted sampler will not stop automatically. We should do it manually.